### PR TITLE
weston-init

### PR DESCRIPTION
**Pull request by thochstein**

### CODE REVIEW

**Review by otavio — APPROVED** [review]

File: recipes-graphics/wayland/weston-init.bbappend [hunk 1]
```diff
@@ -8,64 +8,72 @@ REQUIRED_DISTRO_FEATURES:remove = "${IMX_REQUIRED_DISTRO_FEATURES_REMOVE}"
...
+
+PACKAGECONFIG_USE_G2D                ?= ""
+PACKAGECONFIG_USE_G2D:imxgpu2d       ?= "use-g2d"
+PACKAGECONFIG_USE_G2D:mx8qm-nxp-bsp  ?= ""
+PACKAGECONFIG_USE_G2D:mx8qxp-nxp-bsp ?= ""
+PACKAGECONFIG_USE_G2D:mx8dx-nxp-bsp  ?= ""
+
+USE_G2D_VALUE             = "true"
+USE_G2D_VALUE:mx6-nxp-bsp = "1"
+USE_G2D_VALUE:mx7-nxp-bsp = "1"
```
Comment (angolini) [inline recipes-graphics/wayland/weston-init.bbappend:42]: Could you, please, elaborate why there is "true" and "1"?

File: recipes-graphics/wayland/weston-init.bbappend [hunk 2]
```diff
@@ -8,64 +8,72 @@ REQUIRED_DISTRO_FEATURES:remove = "${IMX_REQUIRED_DISTRO_FEATURES_REMOVE}"
...
+    fi
+
+    if [ "${@bb.utils.contains('PACKAGECONFIG', 'size', 'yes', 'no', d)}" = "yes" ]; then
+        sed -i -e "/^\[shell\]/a size=${SIZE_VALUE}" ${D}${sysconfdir}/xdg/weston/weston.ini
+    fi
+
+    if [ "${@bb.utils.contains('PACKAGECONFIG', 'use-g2d', 'yes', 'no', d)}" = "yes" ]; then
+        sed -i -e "/^\[core\]/a use-g2d=${USE_G2D_VALUE}" ${D}${sysconfdir}/xdg/weston/weston.ini
+    else
+        sed -i -e "/^\[core\]/a #use-g2d=${USE_G2D_VALUE}" ${D}${sysconfdir}/xdg/weston/weston.ini
```
Comment (angolini) [inline recipes-graphics/wayland/weston-init.bbappend:75]: Are line 75 inserting a commented line? I'm not against it, but I would like to double-check if I understand why.

**Review by angolini — APPROVED** [review]
LGTM

File: recipes-graphics/wayland/weston-init.bbappend (hunk 2 above)
Comment (MrCry0) [reply recipes-graphics/wayland/weston-init.bbappend:75]: > Are line 75 inserting a commented line? I'm not against it, but I would like to double-check if I understand why.

@angolini I think it's intentional: if use-g2d is not in PACKAGECONFIG, it is good to add the commented option as a reference.

Comment (thochstein) [reply recipes-graphics/wayland/weston-init.bbappend:75]: Correct.

File: recipes-graphics/wayland/weston-init.bbappend (hunk 1 above)
Comment (thochstein) [reply recipes-graphics/wayland/weston-init.bbappend:42]: The option was updated to a boolean in the weston 11 fork. The option remains as an int for weston 10. I'll update the commit message.

Comment (otavio) [reply recipes-graphics/wayland/weston-init.bbappend:42]: We now use weston 11 everywhere so we could use `true`, no?

Comment (thochstein) [reply recipes-graphics/wayland/weston-init.bbappend:42]: No, weston 11 has fully dropped `fbdev`, and we chose to lock i.MX 6 and 7 on weston 10. See #1537.

Comment (otavio) [reply recipes-graphics/wayland/weston-init.bbappend:42]: My fault. Thx.

**Pull request merged**